src/lib/operators/abstract_operator.cpp

```diff
@@ -258,13 +258,24 @@ std::shared_ptr<const AbstractOperator> AbstractOperator::left_input() const { r
 std::shared_ptr<const AbstractOperator> AbstractOperator::right_input() const { return _right_input; }
 
 void AbstractOperator::set_parameters(const std::unordered_map<ParameterID, AllTypeVariant>& parameters) {
-  DebugAssert(!_executed, "Setting parameters on operators that have already executed is illegal.");
+  Assert(_state == OperatorState::Created, "Setting parameters is allowed for OperatorState::Created only.");
   if (parameters.empty()) return;
   _on_set_parameters(parameters);
   if (left_input()) mutable_left_input()->set_parameters(parameters);
   if (right_input()) mutable_right_input()->set_parameters(parameters);
 }
 
+std::shared_ptr<OperatorTask> AbstractOperator::operator_task() {
+  auto lock = std::unique_lock<std::mutex>(_operator_task_mutex);
```

Comment:
> The rule of thumb is to always use std::lock_guard. But if we need some higher level functionalities, which are available by std::unique_lock, then we should use the std::unique_lock.

http://jakascorner.com/blog/2016/02/lock_guard-and-unique_lock.html

Reply:
Thanks for the hint!